server/client.go

```diff
@@ -2567,6 +2567,8 @@ sendToRoutesOrLeafs:
 		kind := rt.sub.client.kind
 		mh := c.msgb[:msgHeadProtoLen]
 		if kind == ROUTER {
+			// Router (and Gateway) nodes are RMSG. Set here since leafnodes may rewrite.
+			mh[0] = 'R'
```

Comment:
Can we do that also in the other place we use this buffer? That would be gateway.go:2202 (from master)?

Reply:
Will take a look.

Reply:
Fixed once at top of loop.

Reply:
Hopefully a call to processMsgResults() will not creep up in between :-). I think we are safe for now.

Reply:
Well that loop was only sending to gateways. Did not seem to make sense to keep setting same thing in the loop.

Reply:
Yes